File: App/app/src/main/java/com/cinema/ticket_booking/data/local/dao/MovieDao.java
package com.cinema.ticket_booking.data.local.dao;
import androidx.lifecycle.LiveData;
import androidx.room.*;
import com.cinema.ticket_booking.data.local.entity.MovieEntity;
import java.util.List;

@Dao
public interface MovieDao {
    @Query("SELECT * FROM movies ORDER BY title ASC")
    LiveData<List<MovieEntity>> getAllMovies();

    @Query("SELECT * FROM movies WHERE status = :status")
    LiveData<List<MovieEntity>> getMoviesByStatus(String status);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<MovieEntity> movies);

    @Query("DELETE FROM movies")
    void deleteAll();
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/local/dao/MovieDao.kt
package com.cinema.ticket_booking.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.cinema.ticket_booking.data.local.entity.MovieEntity

@Dao
interface MovieDao {
    @Query("SELECT * FROM movies ORDER BY title ASC")
    fun getAllMovies(): LiveData<List<MovieEntity>>

    @Query("SELECT * FROM movies WHERE status = :status")
    fun getMoviesByStatus(status: String): LiveData<List<MovieEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(movies: List<MovieEntity>)

    @Query("DELETE FROM movies")
    fun deleteAll()
}
